SA3/qa_web/views.py

```diff
@@ -76,11 +77,29 @@ def questions(request):
             return render(request, 'qa_web/questionspage.html', context={})
 
 
-def answers(request, id):
-        q = get_object_or_404(Questions, pk=id)
-        return render(request, 'qa_web/answerspage.html', {'currentQuestion': q})
+def answers(request, id_):
+    q = get_object_or_404(Questions, pk=id_)
+    if request.method == 'POST' and 'answer_form' in request.POST: #Update's database when somebody answers a question
+        form = AnswersForm(request.POST)
+        if form.is_valid(): 
+            Answers.objects.create(content=request.POST['content'], owner=request.user, question=q)
+    elif request.method == 'POST' and 'deselect' in request.POST:  #Update's database when somebody deselects best answer.
+        updateAnswer = Answers.objects.filter(question=q, correct_answer=True).last()
+        updateAnswer.correct_answer = False;
+        updateAnswer.save();
+    else: #Update's database when somebody selects a best answer.
+        q_answers = Answers.objects.filter(question=q, correct_answer=False)
+        for answer in q_answers:
+            if request.method == 'POST' and 'select_'+str(answer.id) in request.POST:
+                updateAnswer = Answers.objects.get(id = answer.id)
+                updateAnswer.correct_answer = True;
+                updateAnswer.save();
+    #Get updated answer data.
+    q_answers = Answers.objects.filter(question=q, correct_answer=False)
+    q_best_answer = Answers.objects.filter(question=q, correct_answer=True)
```

Comment:
Differentiating the correct answer from the others could have been done in the template.

Reply:
I had no luck finding the command that allows me to get the answer where correct_answer=True directly from the template. It would need to put the results in another queryset since I need to check if the amount of results with correct answers is 0 (to determine whether to put "Select as best answer buttons").

Reply:
I personally believe handling the context in the backend is actually better as lightweight frontend is better practice. However, instead of passing a `querySet` here you can pass a single object from the `querySet` (correct_answer =true), so that you don't need to "iterate" through it in the template.

Reply:
If you want to handle this in the frontend, just look at template tags. Create a python `.py` file in a folder name templatetag in same directory you find the templates folder and add `{% load name_of_python_file.py %}` at the top of your template. You can now write python functions that are more complex and you call them from your template through template tags. If you're using models then you want to do `from models import (...)` to import the necessary models you want to work with.

Reply:
Passing a single answer instead of a queryset would be a good resolution for this.